src/render/utils/animation.ts

```diff
@@ -54,8 +54,15 @@ export function animateVisualElement(
     } else if (typeof definition === "string") {
         animation = animateVariant(visualElement, definition, options)
     } else {
-        // TODO: Remove any and handle TargetResolver
-        animation = animateTarget(visualElement, definition as any, options)
+        const resolvedDefinition =
+            typeof definition === "function"
+                ? (resolveVariant(
+                      visualElement,
+                      definition,
+                      options.custom
+                  ) as TargetAndTransition)
```

Comment:
Why is this cast needed?

Reply:
We're hijacking a resolution function that could conceivably return `undefined` but in this instance it actually couldn't

Reply:
I think we can resolve that by adding extra type overloads for this function